src/auto-release.ts

```diff
@@ -0,0 +1,671 @@
+
+  constructor(args: ArgsType) {
+    this.args = args;
+    this.logger = createLog(
+      args.veryVerbose ? 'veryVerbose' : args.verbose ? 'verbose' : undefined
+    );
+    this.hooks = makeHooks();
+
+    this.hooks.onCreateRelease.tap('Link onCreateChangelog', release => {
+      release.hooks.onCreateChangelog.tap('Link onCreateChangelog', changelog =>
+        this.hooks.onCreateChangelog.call(changelog)
+      );
+    });
+    this.hooks.onCreateRelease.tap('Link onCreateLogParse', release => {
+      release.hooks.onCreateLogParse.tap('Link onCreateLogParse', logParse =>
+        this.hooks.onCreateLogParse.call(logParse)
+      );
+    });
+
+    env.config();
+  }
+
+  /**
+   * Loads a config from a path, package name, or special `auto-config` pattern
+   *
+   * ex: auto-config-MY_CONFIG
+   * ex: @MY_CONFIG/auto-config
+   *
+   * @param extend Path or name of config to find
+   */
+  public loadExtendConfig(extend: string) {
+    let config: cosmiconfig.Config | ConfigLoader = tryRequire(extend);
+
+    if (!config) {
+      config = tryRequire(`${extend}/auto-config`);
+    }
+
+    if (!config) {
+      config = tryRequire(`auto-config-${extend}`);
+    }
+
+    if (typeof config === 'function') {
+      return (config as ConfigLoader)();
+    }
+
+    return config || {};
+  }
+
+  /**
+   * Load the .autorc from the file system, set up defaults, combine with CLI args
+   * load the extends property, load the plugins and start the git remote interface.
+   */
+  public async loadConfig() {
+    const explorer = cosmiconfig('auto');
+    const result = await explorer.search();
+
+    let rawConfig: cosmiconfig.Config = {};
+
+    if (result && result.config) {
+      rawConfig = result.config;
+    }
+
+    if (rawConfig.extends) {
+      rawConfig = merge(rawConfig, this.loadExtendConfig(rawConfig.extends));
+    }
+
+    this.logger.verbose.success(
+      'Loaded `auto-release` with config:',
+      rawConfig
+    );
+
+    this.semVerLabels = defaultLabels;
+
+    if (rawConfig.labels) {
+      this.semVerLabels = new Map<VersionLabel, string>([
+        ...defaultLabels,
+        ...(Object.entries(rawConfig.labels) as [VersionLabel, string][])
+      ]);
+    }
+
+    this.logger.verbose.success(
+      'Using SEMVER labels:',
+      '\n',
+      this.semVerLabels
+    );
+
+    const skipReleaseLabels = rawConfig.skipReleaseLabels || [];
+
+    if (!skipReleaseLabels.includes(this.semVerLabels.get('skip-release')!)) {
+      skipReleaseLabels.push(this.semVerLabels.get('skip-release')!);
+    }
+
+    const config = {
+      ...rawConfig,
+      ...this.args,
+      versionLabels: this.semVerLabels,
+      skipReleaseLabels
+    };
+
+    this.loadPlugins(config);
+    this.hooks.beforeRun.call(config);
+
+    const repository = await this.getRepo(config);
+    const token =
+      repository && repository.token
+        ? repository.token
+        : await getGitHubToken(config.githubApi);
```

Comment:
We should figure out a way to break this out into the Github plugin

Reply:
This line will probably become a hook.

Reply:
The token stuff should probably move to the git class as well